component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java

```diff
@@ -3392,4 +3392,42 @@ public void testSpaceContainsExternalMembers() throws Exception {
     assertEquals(true, hasExternals);
     tearDownUserList.add(externalUser);
   }
+
+
+  public void testGetCommonSpaces() throws Exception {
+
+    getSpaceInstance(11);
+    Space space1 = getSpaceInstance(12);
+    Space space2 = getSpaceInstance(13);
+    Space space3 = getSpaceInstance(14);
+    Space space4 = getSpaceInstance(15);
+    Space space5 = getSpaceInstance(16);
+
+    User otherUser = organizationService.getUserHandler().createUserInstance("otherUser");
+    organizationService.getUserHandler().createUser(otherUser, false);
+
+    spaceService.addMember(space1,"otherUser");
+    spaceService.addMember(space2,"otherUser");
+    spaceService.addMember(space3,"otherUser");
+    spaceService.addMember(space4,"otherUser");
+    spaceService.addMember(space5,"otherUser");
+
+    ListAccess<Space> resultListCommonSpacesAccessList1 = spaceService.getCommonSpaces("root","otherUser",0,2);
+    Space testSpace1 = resultListCommonSpacesAccessList1.load(0,1)[0];
+    assertEquals(space1,testSpace1);
+    Space testSpace2 = resultListCommonSpacesAccessList1.load(1,1)[0];
+    assertEquals(space2,testSpace2);
+
+    ListAccess<Space> resultListCommonSpacesAccessList2 = spaceService.getCommonSpaces("root","otherUser",2,2);
+    Space testSpace3 = resultListCommonSpacesAccessList2.load(2,1)[0];
+    assertEquals(space3,testSpace3);
+    Space testSpace4 = resultListCommonSpacesAccessList2.load(3,1)[0];
+    assertEquals(space4,testSpace4);
+
+    ListAccess<Space> resultListCommonSpacesAccessList3 = spaceService.getCommonSpaces("root","otherUser",4,2);
+    Space testSpace5 = resultListCommonSpacesAccessList3.load(4,1)[0];
+    assertEquals(space5,testSpace5);
+    assertEquals(1,resultListCommonSpacesAccessList3.load(4,2).length);
```

Comment:
It is better to test the lenght before the content, so that if the length not correspond, you dont have an error when getting [0]
```suggestion
    ListAccess<Space> resultListCommonSpacesAccessList3 = spaceService.getCommonSpaces("root","otherUser",4,2);
    assertEquals(1,resultListCommonSpacesAccessList3.load(4,2).length);
    Space testSpace5 = resultListCommonSpacesAccessList3.load(4,1)[0];
    assertEquals(space5,testSpace5);
```